### PR TITLE
feat:add llm factory

### PR DESCRIPTION
**Pull request by ruanrongman**
This is closed to #322 

## Summary by CodeRabbit

- **New Features**
	- Added `LLMFactory` to the `promptulate` library for dynamic creation of language model instances.
- **Tests**
	- Added tests to validate the functionality of `LLMFactory`.
- **Documentation**
	- Updated the Jupyter notebook to demonstrate the usage of `LLMFactory` for generating text with AI.

### CODE REVIEW

**Comment by Undertone0809** [comment]
You don't need to create a `.py` example if you already have a notebook example. You notebook shouldn't add in core code. Move it to `./exmaple`.

Moreover, you can convert notebook to Markdown and generate in `/docs`

image

**Review by Undertone0809 — CHANGES_REQUESTED** [review]

**Review by Undertone0809 — CHANGES_REQUESTED** [review]

File: promptulate/llms/llm_factory.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        if model_name == "":
+            return CustomLLM(model_config=model_config)
+        else:
+            return CustomLLM(model=model_name, model_config=model_config)
+    else:
+        raise ImportError("not found this platform")
+
+
+class LLMFactory:
+    def __init__(self, platform_type, model_config, **kwargs):
```
[the remark this answers is not in the document]
Comment (ruanrongman) [reply promptulate/llms/llm_factory.py:28]: ok

File: promptulate/__init__.py [hunk 2]
```diff
@@ -24,6 +24,7 @@
 from promptulate.agents.web_agent.agent import WebAgent
 from promptulate.chat import chat
 from promptulate.llms.base import BaseLLM
+from promptulate.llms.llm_factory import LLMFactory
```
[the remark this answers is not in the document]
Comment (ruanrongman) [reply promptulate/__init__.py:27]: ok

File: promptulate/llms/factory.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        if model_name == "":
+            return CustomLLM(model_config=model_config)
+        else:
+            return CustomLLM(model=model_name, model_config=model_config)
+    else:
+        raise ImportError("not found this platform")
+
+
+class LLMFactory:
+    def __init__(
```
Comment (Undertone0809) [inline promptulate/llms/factory.py:28]: Really need initialization method?

**Review by Undertone0809 — APPROVED** [review]

File: promptulate/llms/factory.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+        if model_name == "":
+            return CustomLLM(model_config=model_config)
+        else:
+            return CustomLLM(model=model_name, model_config=model_config)
+    else:
+        raise ImportError("not found this platform")
+
+
+class LLMFactory:
```
Comment (Undertone0809) [inline promptulate/llms/factory.py:27]: LLMFactory API design should be more pythonic. `pne.LLMFactory("qianfan").build()` is too complicated.

**Review by Undertone0809 — CHANGES_REQUESTED** [review]

File: tests/llms/test_factory.py
Comment (Undertone0809) [inline tests/llms/test_factory.py:1]: You should add test_factory.py into Makefile

File: tests/llms/test_factory.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    def test_call(self, mock_post):
+        # Mock the API response
+        mock_response = mock.Mock()
+        mock_response.status_code = 200
+        mock_response.json.return_value = {
+            "choices": [{"text": "[start] This is a test [end]"}]
+        }
+        mock_post.return_value = mock_response
+
+        llm = pne.LLMFactory("qianfan").build()
```
Comment (Undertone0809) [inline tests/llms/test_factory.py:17]: error here

File: promptulate/llms/factory.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    def build(self, platform_type: str, model_name: str = None, model_config=None):
```
Comment (Undertone0809) [inline promptulate/llms/factory.py:28]: build should be a classmethod

Comment (Undertone0809) [inline promptulate/llms/factory.py:28]: config should add data type

**Review by Undertone0809 — CHANGES_REQUESTED** [review]

File: promptulate/llms/factory.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        else:
+            return CustomLLM(model=model_name, model_config=model_config)
+    else:
+        raise ImportError("not found this platform")
+
+
+class LLMFactory:
+    @classmethod
+    def build(
+        self, platform_type: str, model_name: str = None, model_config: {} = None
```
Comment (Undertone0809) [inline promptulate/llms/factory.py:30]: here is cls not self

**Review by Undertone0809 — CHANGES_REQUESTED** [review]

**Pull request merged**